### PR TITLE
CIRC-1364 Add Year to the request JSON structure

### PR DESCRIPTION
**Pull request by sherzodn**
https://issues.folio.org/browse/CIRC-1364

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/RequestRepresentation.java [hunk 1]
```diff
@@ -97,6 +97,8 @@ private static void addInstanceProperties(JsonObject request, Item item) {
       write(instance, "title", item.getTitle());
       write(instance, "identifiers", item.getIdentifiers());
       write(instance, "contributorNames", mapContributorsToNamesOnly(item.getContributors()));
+      write(instance, "publication", item.getPublication());
+      write(instance, "editions", item.getEditions());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/RequestRepresentation.java:101]: Please update tests to cover new fields in GET response

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java [hunk 2]
```diff
@@ -2,19 +2,26 @@
 
 import static java.lang.Boolean.TRUE;
 import static java.time.ZoneOffset.UTC;
+import static org.folio.circulation.domain.representations.InstanceProperties.EDITIONS;
+import static org.folio.circulation.domain.representations.InstanceProperties.PUBLICATION;
+import static org.folio.circulation.support.json.JsonPropertyWriter.write;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (roman-barannyk) [inline src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java:10]: don't import all package

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java [hunk 3]
```diff
@@ -35,6 +42,17 @@ void testExtendedRepresentation() {
 
     assertThat("Extended representation should have a delivery address",
       extendedRepresentation.containsKey("deliveryAddress"), is(true));
+
+    assertThat("Extended representation should have a instance",
```
Comment (roman-barannyk) [inline src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java:46]:
```suggestion
    assertThat("Extended representation should have an instance",
```

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java [hunk 4]
```diff
@@ -35,6 +42,17 @@ void testExtendedRepresentation() {
...
+      extendedRepresentation.containsKey("instance"), is(true));
+
+    JsonObject instance = extendedRepresentation.getJsonObject("instance");
+
+    assertTrue(instance.containsKey(EDITIONS));
+    assertTrue(instance.containsKey(PUBLICATION));
```
Comment (roman-barannyk) [inline src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java:52]: Don't use the constants from the production classes. Create it's own for the tests class

File: src/main/java/org/folio/circulation/domain/RequestRepresentation.java (hunk 1 above)
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/RequestRepresentation.java:101]: Please add the new fields assertion to RequestsAPICreationTests and RequestsAPIRetrievalTests

Comment (sherzodn) [reply src/main/java/org/folio/circulation/domain/RequestRepresentation.java:101]: done

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java (hunk 2 above)
Comment (sherzodn) [reply src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java]: done

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java (hunk 3 above)
Comment (sherzodn) [reply src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java]: done

File: src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java (hunk 4 above)
Comment (sherzodn) [reply src/test/java/org/folio/circulation/domain/RequestRepresentationTests.java:52]: done

File: src/main/java/org/folio/circulation/domain/RequestRepresentation.java (hunk 1 above)
Comment (sherzodn) [reply src/main/java/org/folio/circulation/domain/RequestRepresentation.java:101]: done

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**